src/dowel/csv_output.py

```diff
@@ -19,6 +20,7 @@ def __init__(self, file_name):
         self._fieldnames = None
         self._warned_once = set()
         self._disable_warnings = False
+        self._df = pd.DataFrame()  # Pandas DataFrame, used to update header
```

Comment:
This comment doesn't seem to be very helpful for what's going on here. The dataframe actually contains all data logged so far, right?

Reply:
Yes, I'll make it more clear.